GCRCatalogs/dc2_coadd.py

```diff
@@ -5,23 +5,77 @@
 import os
 import re
 import warnings
+
 import numpy as np
-import tables
 import pandas as pd
+import tables
 from GCR import BaseGenericCatalog
 
 __all__ = ['DC2CoaddCatalog']
 
+FILE_PATTERN = r'merged_tract_\d+\.hdf5'
+GROUP_PATTERN = r'coadd_\d+_\d\d$'
+
+
+def calc_cov(ixx_err, iyy_err, ixy_err):
+    """Calculate the covariance between three arrays of second moments
+
+    Args:
+        ixx_err (float): The error in the second moment Ixx
+        iyy_err (float): The error in the second moment Iyy
+        ixy_err (float): The error in the second moment Ixy
+
+    Returns:
+        Elements of the covariance matrix ordered as
+        [ixx * ixx, ixx * ixy, ixx * iyy, ixy * ixy, ixy * iyy, iyy * iyy]
+    """
+
+    # If need be, this can be generalized using a mesh grid
+    out_data = np.array([
+        ixx_err * ixx_err,
+        ixx_err * ixy_err,
+        ixx_err * iyy_err,
+        ixy_err * ixy_err,
+        ixy_err * iyy_err,
+        iyy_err * iyy_err
+    ])
```

Comment:
This calculation is not correct.  We need the original covariance values calculated by the pipeline.  These are dependent on the counts in the pixels; we can't reconstruct them later solely from the diagonal terms.